### PR TITLE
Start ReceiveRPC Override + Improvements

### PR DESCRIPTION
**Pull request by Ultradragon005**
did all the ones in syncroleskillreader as it was simple, and @NikoCat233 may do the rest.
Basically ready to merge, I'm just adding/changing stuff as I see fit rn

Did these points for all roles now.
- [x] Standarized setup for all roles
- [x]  Replace List with HashSet where is possible
- [x]  Make List, HashSet, Dictionary as readonly and in Init() use Clear() instead of = []; where is possible
- [x] RpcMurderPlayerV3 => RpcMurderPlayer
- [x] Add RpcCheckAndMurder & CheckForInvalidMurdering in some places of the code

### CODE REVIEW

**Comment by Ultradragon005** [comment]
Standard setup should be done for all roles nowm I think I already did all neutrals.

File: Modules/ExtendedPlayerControl.cs [hunk 1]
```diff
@@ -640,40 +640,18 @@ public static void RpcExileV2(this PlayerControl player)
...
-
-        if (target.Is(CustomRoles.Susceptible))
-            Susceptible.CallEnabledAndChange(target);
-
-        if (killer.PlayerId == target.PlayerId && killer.shapeshifting)
+        if (killer.PlayerId == target.PlayerId && killer.shapeshifting) 
         {
             _ = new LateTask(() => { killer.RpcMurderPlayer(target, true); }, 1.5f, "Shapeshifting Suicide Delay");
             return;
         }
```
Comment (Ultradragon005) [inline Modules/ExtendedPlayerControl.cs:647]: was this something important? Should jsut simply be moved to rpcmurder then.

Comment (Tommy-XL) [reply Modules/ExtendedPlayerControl.cs:647]: No, it's not needed
image

**Comment by Ultradragon005** [comment]
Ig this is finished

**Pull request merged**